consensus/istanbul/uptime/uptime_test.go

```diff
@@ -0,0 +1,124 @@
+package uptime
+
+import (
+	"math/big"
+	"reflect"
+	"testing"
+)
+
+func TestGetUptimeMonitoringWindow(t *testing.T) {
+	type args struct {
+		epochNumber        uint64
+		epochSize          uint64
+		lookbackWindowSize uint64
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  uint64
+		want1 uint64
+	}{
+		{"tally on first epoch", args{1, 10, 2}, 1 + 2, 9},
+		{"tally on second epoch", args{2, 10, 2}, 11 + 2, 19},
+		{"lookback window too big", args{1, 10, 10}, 11, 9},
```

Comment:
I'm skeptical about including a case with an invalid combination of lookback window size and epoch size.  Is this a case that we are supposed to support or that should lead to a fatal error somewhere?  If it's the latter, we should test elsewhere does it does lead to a fatal error, and not test it here.

Reply:
i'll review this on when finishing https://github.com/celo-org/celo-blockchain/pull/1136